### PR TITLE
Issue #2664712: Removing the redundant entity part from entity create…

### PR DESCRIPTION
**Pull request by emclaughlin1215**
… action IDs and making sure all tests pass.

See https://www.drupal.org/node/2664712

### CODE REVIEW

**Comment by emclaughlin1215** [comment]
Looks like the checks failed due to this bug: https://www.drupal.org/node/2659954

File: src/Plugin/RulesAction/EntityCreate.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   public function refineContextDefinitions() {
     if ($type = $this->entityTypeId) {
-      $data_type = "entity:$type";
+      $data_type = "$type";
```
Comment (klausi) [inline src/Plugin/RulesAction/EntityCreate.php]: The $data_type variable is now not needed anymore and should be removed.

File: src/Plugin/RulesAction/EntityCreateDeriver.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ public function getDerivativeDefinitions($base_plugin_definition) {
         continue;
       }
 
-      $this->derivatives["entity:$entity_type_id"] = [
+      $this->derivatives["$entity_type_id"] = [
```
Comment (klausi) [inline src/Plugin/RulesAction/EntityCreateDeriver.php]: quotes not necessary

File: src/Plugin/RulesAction/EntityCreateDeriver.php [hunk 3]
```diff
@@ -101,7 +101,7 @@ public function getDerivativeDefinitions($base_plugin_definition) {
           $context_definition->setAssignmentRestriction(ContextDefinition::ASSIGNMENT_RESTRICTION_INPUT);
         }
 
-        $this->derivatives["entity:$entity_type_id"]['context'][$field_name] = $context_definition;
+        $this->derivatives["$entity_type_id"]['context'][$field_name] = $context_definition;
```
Comment (klausi) [inline src/Plugin/RulesAction/EntityCreateDeriver.php]: quotes not necessary

File: tests/src/Integration/Action/EntityCreateTest.php [hunk 4]
```diff
@@ -143,7 +143,7 @@ public function testRefiningContextDefinitions() {
     $this->action->refineContextDefinitions();
     $this->assertEquals(
       $this->action->getProvidedContextDefinition('entity')
-        ->getDataType(), 'entity:test:bundle_test'
+        ->getDataType(), 'test:bundle_test'
```
Comment (klausi) [inline tests/src/Integration/Action/EntityCreateTest.php]: this is wrong, the provided data type must be entity:test:bundle_test. We should only change the plugin ID.

**Comment by klausi** [comment]
The git commits are not connected to you user account, you probably need to add that email address to github.

**Comment by klausi** [comment]
merged.

**Pull request closed without merging**